### PR TITLE
remove all remaining uses of QuantumTape in pennylane

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
Follow #3454. There was some lingering usage of `QuantumTape` other than `with tape`, and they too must be removed before `QuantumTape` can be removed.

**Description of the Change:**
Removes all remaining uses of `QuantumTape` in pennylane

**Benefits:**
There is less `QuantumTape` in the world.

**Possible Drawbacks:**
N/A

## EDIT
As requested, I've put back all docstrings. This PR now only makes very minimal changes, refactors, cleanups, etc.

### CODE REVIEW

File: pennylane/transforms/qcut/montecarlo.py [hunk 1]
```diff
@@ -50,14 +50,14 @@
 
 @batch_transform
 def cut_circuit_mc(
-    tape: QuantumTape,
+    tape: QuantumScript,
```
Comment (josh146) [inline pennylane/transforms/qcut/montecarlo.py:53]: similarly here

File: pennylane/transforms/transpile.py [hunk 2]
```diff
@@ -11,13 +11,13 @@
...
 from pennylane.queuing import QueuingManager
-from pennylane.tape import QuantumTape
+from pennylane.tape import QuantumScript
 from pennylane.transforms import qfunc_transform
 from pennylane.wires import Wires
 
 
 @qfunc_transform
-def transpile(tape: QuantumTape, coupling_map: Union[List, nx.Graph]):
+def transpile(tape: QuantumScript, coupling_map: Union[List, nx.Graph]):
```
Comment (josh146) [inline pennylane/transforms/transpile.py:20]: and this one :)

Comment (josh146) [reply pennylane/transforms/transpile.py:20]: @trbromley I might tag you for the qcut functions

**Comment by timmysilv** [comment]
I have put back the docstrings and type-hints everywhere to use `QuantumTape`. I'd like to re-iterate in case this wasn't clear, those docstrings are now ***lying***. All QuantumTapes are QuantumScripts, but not all QuantumScripts are QuantumTapes. Fortunately, PennyLane now only uses QuantumScripts internally. This means that if you input a QuantumTape or a QuantumScript into any of these functions, they should work fine. The confusing thing is that some functions will output a pure QuantumScript, contrary to user expectations (because of the docstring/type-hints that lied to them)

**Comment by timmysilv** [comment]
not adding a changelog entry, there isn't really anything being changed here (at least, nothing that isn't already in progress in other PRs)

**Review by lillian542 — APPROVED** [review]
Seems done to me :)

**Pull request merged**